server/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -413,7 +413,8 @@ object CSRF {
           case Right(dec) =>
             // The `.drop(1)` removes the first empty line from the body value of the field.
             dec.parts
-              .find(_.headers.exists(_.value.contains(s"""name="$fieldName"""")))
+              .find(_.headers.exists(h =>
+                h.is(headers.`Content-Disposition`) && h.value.contains(s"""name="$fieldName"""")))
```

Comment:
Does it not parse the `name` into the `parameters`?

Reply:
Sorry, was a bit in a rush yesterday. I cleaned it up a bit.
However I still have trouble with it in integration. I published locally and used in an exisiting project and there it doesn't seem to work.
After the weekend I'll dig deeper into the issue and try to reproduce that.

Reply:
My guess is that it's related to that consuming the stream twice.  This seems like a difficult problem to handle in a streaming library: we need to parse the stream to find the token, and then buffer that stream so the app can handle it, which fights against the nature of streaming in the first place.  It's doable, but it's going to come with some nasty space tradeoffs.